src/app/app.component.ts

```diff
@@ -1,9 +1,34 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { Subject } from 'rxjs';
+import { filter, takeUntil, tap } from 'rxjs/operators';
+
+import { EventsService } from 'src/app/modules/core/services/events.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
 })
-export class AppComponent {
+export class AppComponent implements OnInit, OnDestroy {
   title = 'prysm-web-ui';
+  constructor(
+    private router: Router,
+    private eventsService: EventsService,
+  ) { }
+  private destroyed$$ = new Subject<void>();
+
+  ngOnInit(): void {
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
```

Comment:
If we use reactive programming, we can use the `filter` operator to only receive navigation end events, removing the need for `if` conditions in the service